### PR TITLE
Bugfix/mar 403 search field

### PR DESCRIPTION
**Pull request by drgoshm**

### CODE REVIEW

File: src/pages/MyTokens/NFTPage.tsx [hunk 1]
```diff
@@ -136,13 +138,13 @@ export const NFTPage = () => {
...
-      if (prices?.minPrice && prices?.maxPrice) {
+      if (prices?.minPrice || prices?.maxPrice) {
         if (!token.price) {
           filteredByPrice = false;
         } else {
           const tokenPrice = new BN(token.price);
-          const minPrice = new BN(fromStringToBnString(prices.minPrice, api?.market?.kusamaDecimals));
-          const maxPrice = new BN(fromStringToBnString(prices.maxPrice, api?.market?.kusamaDecimals));
+          const minPrice = new BN(fromStringToBnString(prices.minPrice || '0', api?.market?.kusamaDecimals));
+          const maxPrice = prices.maxPrice ? new BN(fromStringToBnString(prices.maxPrice, api?.market?.kusamaDecimals)) : BN_MAX_INTEGER;
```
Comment (SergeyOkishev) [inline src/pages/MyTokens/NFTPage.tsx:147]: better not to send that field in request and let back-end resolve it. Force putting max_int is a bad practice.

Comment (drgoshm) [reply src/pages/MyTokens/NFTPage.tsx:147]: There are no backend requests. These filters are only used on the frontend.

File: src/api/chainApi/unique/NFTController.ts [hunk 2]
```diff
@@ -28,69 +28,62 @@ class UniqueNFTController implements INFTController<NFTCollection, NFTToken> {
...
-          const ipfsJson = JSON.parse(decodedConstData.ipfsJson as string);
-
-          imageUrl = `${IPFSGateway}/${ipfsJson.ipfs}`;
-        } else {
-          imageUrl = await getTokenImage(collectionInfo, tokenId);
-        }
+      if (collection.schemaVersion.isUnique && decodedConstData.ipfsJson) {
+        const ipfsJson = JSON.parse(decodedConstData.ipfsJson as string) as { ipfs: string };
+        imageUrl = `${IPFSGateway}/${ipfsJson.ipfs}`;
+      } else if (collection.schemaVersion.isImageURL) {
```
Comment (SergeyOkishev) [inline src/api/chainApi/unique/NFTController.ts:55]: tokenURI is missing

File: src/api/chainApi/utils/imageUtils.ts [hunk 3]
```diff
@@ -31,10 +30,11 @@ const fetchTokenImage = async (
   return '';
 };
 
-export const getTokenImage = async (collection: NFTCollection, tokenId: number): Promise<string> => {
-  if (collection.schemaVersion === 'ImageURL' || collection.schemaVersion === 'TokenURI') {
-    return getTokenImageUrl(hex2a(collection.offchainSchema), tokenId);
+export const getTokenImage = async (collection: UpDataStructsCollection, tokenId: number): Promise<string> => {
+  if (collection.schemaVersion.isImageURL) {
```
Comment (SergeyOkishev) [inline src/api/chainApi/utils/imageUtils.ts:34]: tokenURI

**Review by SergeyOkishev — APPROVED** [review]

**Pull request merged**